src/commands/args.ts

```diff
@@ -0,0 +1,49 @@
+import yargs from "yargs";
+
+export const indexCommandArgs = (yarg: yargs.Argv) => {
+  return yarg
+    .positional("filenames", {
+      describe: "WARC file(s) to index",
+      type: "string",
+      array: true,
+      demandOption: "true",
+    })
+    .option("fields", {
+      alias: "f",
+      describe: "fields to include in index",
+      type: "string",
+    });
+};
+
+//export type IndexCommandArgs = Awaited<typeof indexCommandArgs.argv>;
+// todo: fix types?
+export type IndexCommandArgs = any;
```

Comment:
@jlarmstrongiv @chelkyl Had to change the `indexCommandArgs` to actually be a function since it operates on a local yarg object (maybe there's a different way).
It all works, but not sure what the `IndexCommandArgs` `CdxIndexCommandArgs` should be..
`ReturnType<typeof indexCommandArgs>` seems close, but not quite.
Any thoughts?

Reply:
(Could also just leave it as `any` for now and come back to this later)

Reply:
Will come back to this later.

Reply:
I took a quick look at that today, but didn’t quite get it working either.  I’ll dive deeper to fix it later